src/index.ts

```diff
@@ -32,7 +34,11 @@ export type HeadlessConfig = AnswersConfig & {
    * The verticalKey associated with the vertical to manage. If none is provided,
    * Answers Headless will manage universal search.
    */
-  verticalKey?: string
+  verticalKey?: string,
+  /**
+   * {@inheritDoc HeadlessAdditionalHttpHeaders}
+   */
+  additionalHttpHeaders?: HeadlessAdditionalHttpHeaders
```

Comment:
@cea2aj, Is there anyway we can make this property hidden? I don't think there's much value in exposing it to users of Headless at the moment. Headless operates at a higher level of abstraction than Core, so it's reasonable to assume they won't need to know anything about this. Only our libraries will leverage this property internally. 

Depending on what other Headers we make available in the future, we could make this public. But, to start with, I think having it be hidden in the docs could be good. I don't care that it's still visible in the Constructor. If you're building Headless yourself (not through `provideCore`) you're a power user anyway.

Reply:
Unfortunately because `HeadlessConfig` is a type and not an interface, we can't make this particular property internal, however since `HeadlessAdditionalHttpHeaders` is an interface so we could make its own properties internal. When when mark a property internal, it's excluded from the docs and the type bundle.

Reply:
Currently API Extractor isn't trimming the type bundle, so if we do add an `@internal` tag, we'd need to disable the `untrimmedFilePath` and enable the `publicTrimmedFilePath`

Reply:
Alternatively, we could remove `additionalHttpHeaders` from `HeadlessConfig` altogether and we supply the Headless Agent info behind the scenes. That would let us avoid making these additional interfaces.